backend/service/service_proposal.go

```diff
@@ -133,45 +171,164 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
-			passThreshold, vetoThreshold, participation, err := lcd.GetPassVetoThresholdAndParticipationMinDeposit(propo.Type)
+		tmp.Level = l
+
+		proposals = append(proposals, tmp)
+	}
+	return proposals
+}
+
+func getDelegationsByVoter(curValidators map[string]*vo.ValidatorGovInfo, curDelegators map[string]*vo.DelegatorGovInfo, vote document.PVote, allBondedValidators []document.Validator) float64 {
+	var totalVotedPower float64
+	curDelegators[vote.Voter] = &vo.DelegatorGovInfo{
+		Option:      vote.Option,
+		Address:     vote.Voter,
+		Time:        vote.Time,
+		IsValidator: false,
+	}
+	curDelegator := curDelegators[vote.Voter]
+	valAddr := utils.Convert(conf.Get().Hub.Prefix.ValAddr, vote.Voter)
+	delegations := lcd.GetDelegationsByDelAddr(vote.Voter)
+	for _, delegation := range delegations {
+		if delegation.ValidatorAddr == valAddr {
+			curDelegator.IsValidator = true
+			curDelegator.ValAddr = valAddr
+		}
+		var votingPower float64
+		delegationShares, err := utils.ParseStringToFloat(delegation.Shares)
+		if err != nil {
+			logger.Error("ParseInt delegationShares have error")
+		} else {
+			votingPower = computedVotingPower(delegationShares, allBondedValidators, delegation.ValidatorAddr)
+		}
+		totalVotedPower = totalVotedPower + votingPower
+		curDelegator.DelVotingPower = curDelegator.DelVotingPower + votingPower
+
+		if _, ok := curValidators[delegation.ValidatorAddr]; !ok {
+			for _, v := range allBondedValidators {
+				if v.OperatorAddress == delegation.ValidatorAddr {
+					delegatorShares, _ := utils.ParseStringToFloat(v.DelegatorShares)
+					tokens, _ := utils.ParseStringToFloat(v.Tokens)
+					curValidators[delegation.ValidatorAddr] = &vo.ValidatorGovInfo{
+						Address:   delegation.ValidatorAddr,
+						DelShares: delegatorShares,
+						Tokens:    tokens,
+					}
+					break
+				}
+			}
+		}
+		curValidator := curValidators[delegation.ValidatorAddr]
+		curValidator.DelDeductionShares = curValidator.DelDeductionShares + delegationShares
+	}
+	return totalVotedPower
+}
 
-			if err != nil {
-				logger.Error("GetThresholdAndParticipationMinDeposit", logger.String("err", err.Error()), logger.String("param", propo.Type))
+func computedVotingPower(delegationShares float64, allBondedValidators []document.Validator, valAddr string) float64 {
+	for _, v := range allBondedValidators {
+		if v.OperatorAddress == valAddr {
+			tokens, err := utils.ParseStringToFloat(v.Tokens)
+			if err == nil {
+				delegatorShares, err := utils.ParseStringToFloat(v.DelegatorShares)
+				if err == nil {
+					votingPower := (tokens / delegatorShares) * delegationShares
+					return votingPower
+				}
 			}
-			l.GovParam = vo.GovParam{
-				PassThreshold: passThreshold,
-				VetoThreshold: vetoThreshold,
-				Participation: participation,
+		}
+	}
+	return 0
+}
+
+func computedValidatorsPower(curValidators map[string]*vo.ValidatorGovInfo, curDelegators map[string]*vo.DelegatorGovInfo, total float64) float64 {
```

Comment:
there is no need to return `totalVotingPower`

Reply:
deleted it